homeassistant/components/yolink/switch.py

```diff
@@ -118,8 +174,16 @@ def update_entity_state(self, state: dict[str, Any]) -> None:
 
     async def call_state_change(self, state: str) -> None:
         """Call setState api to change switch state."""
-        await self.call_device_api("setState", {"state": state})
-        self._attr_is_on = self.entity_description.value(state)
+        params: dict[str, str | int] = {"state": state}
+        if self.entity_description.device_type == ATTR_DEVICE_MULTI_OUTLET:
+            params["chs"] = 1 << self.entity_description.plug_index
+        await self.call_device_api("setState", params)
+        cur_state: str | list[str | None] = state
+        if self.entity_description.device_type == ATTR_DEVICE_MULTI_OUTLET:
+            state_arr: list[str | None] = [None] * 5
+            state_arr[self.entity_description.plug_index] = state
+            cur_state = state_arr
```

Comment:
OK, so this builds an array which the `value` lambda can then fetch the right value from.
It seems like a quite roundabout way of doing things, can you try to come up with something simpler?

Maybe, instead of the `value` lambdas, add a helper function which takes as input the `state`, the `device_type`, and the `plug_index` and does the needful based on that? If the `state` is a string, just use that, and if it's a list, pick the right index.

Reply:
This is for compatibility with `update_entity_state` method which update entity state via mqtt data push

Reply:
Yes, I understand, but making a temporary list is quite useless.
I would just remove the `value` property from `YoLinkSwitchEntityDescription`, since all the switches anyway check the same value and replace that with calling a function similar to this:

```py
def _get_state(value: str | list[str], plug_index: int) -> str | None:
    if isinstance(value, list):
        return value[plug_index] == "open"
    return value == "open" if value is not None else None
```